doc/dynamic-data.md

```diff
@@ -24,6 +24,51 @@ This is the information available in each provider.
 | OpenStack-Metadata | ✓          | ✓              | ✓             |                |               |
 | Vagrant-Virtualbox | ✓          | ✓              |               |                |               |
 
+## Custom metadata providers
+
+`ct` also supports custom metadata providers. To use the `custom` platform, modify the coreos-metadata service unit to execute your own custom metadata fetcher. The custom metadata fetcher must write an environment file to `/run/metadata/coreos` which defines all the above environment variables used in the Container Linux Config. The environment variables are the same as in the Container Linux Config, but prefixed with `COREOS_`.
```

Comment:
`COREOS_CUSTOM_` now.

What happens if only some variables are defined?  There might not be all four of public/private IPv4/IPv6.

Reply:
Oops, will fix that.

`ct` has no way of knowing what the custom metadata service will be able to provide, so it always allows them. I just tested and systemd will resolve an undefined environment variable to an empty string.

Reply:
My intent with that sentence was to say that everything the user uses in their CLC must written out by the custom metadata service. What about:

"The custom metadata fetcher must write an environment file to `/run/metadata/coreos` defining every `COREOS_CUSTOM_*` environment variable used in the specified Container Linux Config"

Reply:
Changed to that.